File: biddo_client/lib/core/models/asset.dart
class Asset {
  String id;
  String path;
  int size;

  String? fullPath;

  DateTime? createdAt;
  DateTime? updatedAt;

  Asset({
    required this.id,
    required this.path,
    required this.size,
    this.fullPath,
    this.createdAt,
    this.updatedAt,
  });

  static Asset fromJSON(dynamic data) {
    return Asset(
      id: data['id'],
      path: data['path'],
      size: data['size'],
      fullPath: data['fullPath'],
      createdAt: DateTime.parse(data['createdAt']),
      updatedAt: DateTime.parse(data['updatedAt']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'path': path,
      'size': size,
      'fullPath': fullPath,
      'createdAt': createdAt?.toString(),
      'updatedAt': updatedAt?.toString(),
    };
  }
}

File: core/domain/asset.ts
export class Asset {
  id: string
  path: string
  size: number

  createdAt: Date
  updatedAt: Date

  constructor(params: Record<string, unknown> = {}) {
    this.id = params.id as string
    this.path = params.path as string
    this.size = params.size as number
    this.createdAt = params.createdAt as Date
    this.updatedAt = params.updatedAt as Date
  }

  static fromJSON(data: Record<string, unknown> = {}) {
    return new Asset({
      id: data.id,
      path: data.path,
      size: data.size,
      createdAt: data.createdAt ? new Date(data.createdAt.toString()) : null,
      updatedAt: data.updatedAt ? new Date(data.updatedAt.toString()) : null,
    })
  }
}
